components/Services/Card.tsx

```diff
@@ -1,24 +1,35 @@
 'use client';
-import { useState } from 'react';
+import { useRef, useState } from 'react';
 import CardImage from './CardImage';
+import useIntersectionObserver from '@/hooks/useInterSectionObserver';
 
 type ServiceCardProps = {
 	title: string;
 	darkIcon: string;
 	lightIcon: string;
 	text: string;
+	index: number;
 };
 
 export default function ServiceCard({
 	title,
 	darkIcon,
 	lightIcon,
 	text,
+	index,
 }: ServiceCardProps) {
 	const [isHover, setIsHover] = useState<boolean>(false);
+	const ref = useRef<HTMLDivElement>(null);
+
+	const stylesOdd = ['animate-fade-left', 'animate-once', 'md:animate-fade-up'];
+	const styleven = ['animate-fade-right', 'animate-once', 'md:animate-fade-up'];
+
+	useIntersectionObserver(ref, index % 2 !== 0 ? stylesOdd : styleven);
+
 	return (
 		<div
-			className='ease group mx-auto aspect-square w-80 rounded-lg bg-white-800 px-5 py-7 shadow-lg transition-all duration-500 hover:translate-y-10 hover:bg-primary-light dark:bg-black-200 dark:shadow-[#1F2E48] dark:hover:bg-primary-dark hover:dark:shadow-primary-dark sm:w-72'
+			ref={ref}
+			className='ease group mx-auto aspect-square w-80 rounded-lg bg-white-800 px-5 py-7 opacity-0 shadow-lg transition-all duration-500 hover:translate-y-10 hover:bg-primary-light dark:bg-black-200 dark:shadow-[#1F2E48] dark:hover:bg-primary-dark hover:dark:shadow-primary-dark sm:w-72'
 			onMouseEnter={() => setIsHover(true)}
 			onMouseLeave={() => setIsHover(false)}
 		>
```

Comment:
What are we trying to do using these mouse event listeners?

Reply:
i want to change the image when hover, only solution that i found

Reply:
https://play.tailwindcss.com/71EoMp4ILN

Try using CSS instead! You can get rid of the black and white images this way, as well. You can simply apply grayscale to the images with CSS unless there is a hover.